src/components/TodoCard.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useRef, useEffect } from "react";
+import { MdEditCalendar } from "react-icons/md";
+import {
+  BsCalendarCheckFill,
+  BsTrashFill,
+  BsCalendarXFill,
+} from "react-icons/bs";
+import clsx from "clsx";
+import { TodoModal } from "./TodoModal";
+import * as api from "../api";
+import toast from "react-hot-toast";
+import { toastError, toastSuccess } from "../util/constant";
+import { useTodo } from "../context";
+
+export const TodoCard = ({ className, todo }) => {
+  const { syncTodosWithServer } = useTodo();
+
+  const initialRender = useRef(true);
+
+  const [showModal, setShowModal] = useState(false);
+  const [isComplete, setIsComplete] = useState(false);
+
+  useEffect(() => {
+    !initialRender.current && toggleTodoStatus();
+    initialRender.current = false;
+  }, [isComplete]);
+
+  const toggleTodoStatus = async () => {
+    try {
+      const { status } = await api.updateTodo(todo._id, {
+        ...todo,
+        isComplete,
+      });
+      if (status !== 200) return;
+      syncTodosWithServer();
+    } catch (error) {
+      toast.error(toastError.GENERAL);
+    }
+  };
+
+  const deleteTodoRequest = async () => {
+    try {
+      const { status, data } = await api.deleteTodo(todo._id);
+      if (status !== 200) return;
+      syncTodosWithServer();
+      console.log(data);

```

Comment:
Console can be removed.. if not required

Reply:
done, thankyou for pointing out